src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -363,6 +366,8 @@ public function load(array $configs, ContainerBuilder $container)
         if (!empty($config['test'])) {
             $loader->load('test.php');
 
+            $container->getDefinition('clock')?->setClass(MockClock::class);
```

Comment:
The constructors of MockClock and NativeClock have distinct signatures.
The service should be completely redefined. Eventually with an other name: `clock.test`.